Latest Posts block: remove `columns-x` class for list view

Soean opened the pull request:
## Description
This PR removes the `columns-x` class for list view in the Latest Posts block. This class is only necessary for grid view.
Fixes #4015

jahvi committed:
Allow for multi line text in button block

Fixes #2249

tfrommen committed:
Add aria-label to code and HTML block textarea elements. #2027
Also, add placeholder to HTML block.

tfrommen committed:
Update aria-label, and remove (superfluous) placeholder. #2027

stevehenty committed:
Fix the radio control checked attribute

BE-Webdesign committed:
Add full coverage for REQUEST_POST_UPDATE_SUCCESS effect.

Adds test cases to fully cover the REQUEST_POST_UPDATE_SUCCESS effect.

BE-Webdesign committed:
Add tests for ColorPalette component.

BE-Webdesign committed:
Add tests for Editable.getSettings()

BE-Webdesign committed:
Add tests for Editable.adaptFormatter()

Commit:
Add editor heading for screenreaders

This is round 2 of #3801.

It mitigates/fixes #503 and fixes #2024.

What remains to be done in this branch is:

- Make sure the text is translatable, right now it's hardcoded to "Edit Post"
- Make sure the text is contextual (so it shows Edit Page, Edit CPT etc)

Commit:
Make post title contextual and translatable

This just uses the stock WordPress objects.

jorgefilipecosta committed:
Implemented isInlineForTag mechanism

This mechanism allows treating elements as inline if they are being pasted in specific tags. So for example in heading blocks we can interpret headings as inline content and on lists we can interpret lists as inline content.

Commit:
Remove NVDA hack.

youknowriad committed:
Meta Boxes: Show the "advanced" metaboxes location

afercia committed:
Try block advanced settings dynamic menu item.

Rahmon committed:
Fix blocks when useOnce is true

afercia committed:
Add speak messages.

gziolo committed:
Blocks: Move logic for generating class name to BlockEdit component

gziolo committed:
Blocks: Add test for generated class name with BlockEdit

BE-Webdesign committed:
Updating effects to make use of matchers instead of explicit matching.

aduth committed:
Block API: Refactor supportHTML as supports property

aduth committed:
Components: Add withState higher-order component

aduth committed:
Blocks: Port HTML block to use withState

Soean committed:
Show column class only for grid

Soean committed:
Remove columns class for lists also in server-side rendering

bradyvercher commented on blocks/library/latest-posts/index.php:
```diff
@@ -52,7 +52,7 @@ function gutenberg_render_block_core_latest_posts( $attributes ) {
 	}
 
-	if ( isset( $attributes['columns'] ) ) {
+	if ( isset( $attributes['columns'] ) && 'grid' === $attributes['layout'] ) {
```
Based on the surrounding code, it looks like this should check to make sure the `layout` key exists before checking its value.

Soean replied:
I think we can remove all isset() from this file, because the attributes were moved to the server in #2529
cc: @aduth

aduth replied:
We might still want the `'grid' === $attributes['layout']` check though since based on the behavior in the editor, it seems that columns are only relevant for the grid layout.

Soean committed:
Changes buttonProps to html attributes

Rahmon committed:
Fix when block hasn't been used yet in the post

noisysocks committed:
Refactor Reusable Blocks controller to use WP_REST_Posts_Controller

We can remove a lot of code by subclassing WP_REST_Posts_Controller,
which gives us methods for creating, reading, updating and deleting
posts (in this case, blocks) for free.

This also adds support for DELETEing a block to the API, and renames the
'name' field to 'title'.

Commit:
Try fainter disabled state

This changes the opacity of a disabled button to be .3 opacity from .6 previously. This fixes #3928

gziolo committed:
Tests: Refactor tests for REQUEST_POST_UPDATE_SUCCESS effects

gziolo committed:
Merge pull request #3820 from WordPress/add/test/editor/effects/request-post-update-success

Add full coverage for REQUEST_POST_UPDATE_SUCCESS effect.

gziolo committed:
Extensibility: Defer applying filters for component until it is about to be mounted (#4002)

Commit:
Merge pull request #4033 from WordPress/try/fainter-disabled-state

Try fainter disabled state

gziolo committed:
Editor: Move generatedClassName logic out of BlockListBlock

aduth committed:
Merge pull request #3988 from Rahmon/fix/3968

Fix blocks when useOnce is true

aduth committed:
Merge pull request #4016 from WordPress/add/with-state

Components: Add withState higher-order component

aduth committed:
Merge pull request #4014 from WordPress/update/supports-html

Block API: Refactor supportHTML as supports property

jorgefilipecosta committed:
Replaced getLatestPosts usage (wp.api.collections.Posts wrapper) with withAPIData HoC in latestPosts.

Rahmon committed:
Minor refactoring: change from "find" to "some" function (#4052)

afercia committed:
Merge pull request #3982 from WordPress/try/block-settings-menu-item-dynamic

Make the block settings menu item dynamic

youknowriad committed:
Framework: Introduce the data module (#3832)

youknowriad committed:
Data Module: Restrict the state access to the module registering the reducer only

gziolo committed:
Blocks: Make sure Reusable Blocks get class name apllied properly

youknowriad committed:
Extensibility: Respect the 'enter_title_here' hook

gziolo committed:
Editor: Wrap invalid block preview with key

jorgefilipecosta committed:
Added tag groups for headings and lists to the inlineWhitelistTagGroups mechanism.

This improves the behaviour of paste in already existing list and heading blocks.

jorgefilipecosta committed:
Added test cases to isInlineContent to take into account the new isInlineForTag mechanism.

youknowriad committed:
Blocks: Add the missing alignments classNames to the Cover Image Block's markup

gziolo committed:
Merge pull request #4009 from WordPress/update/generated-class-name-edit-block

Blocks: Move logic for generating class name to BlockEdit component

Soean committed:
Merge pull request #4029 from WordPress/update/block-audio-button-attr

Audio Block: Changes edit buttonProps to html attributes

jorgefilipecosta committed:
Added querystringify npm dependency.

jorgefilipecosta committed:
Moved Categories Block edit to an external file.

An intermediary step to use withAPIData.

ntwb committed:
Update Jest to v22.0.1

ntwb committed:
Update Jest snapshots

ntwb committed:
Update babel-jest to v22.0.1

ntwb committed:
Update eslint-plugin-jest to v21.5.0

gziolo committed:
Merge pull request #4073 from WordPress/update/jest

Update to Jest v22.0.1

ntwb committed:
Recreate `package-lock.json` using Node.js 8.9.3 & npm 5.6.0 (#4075)

jorgefilipecosta committed:
Replaced getCategories usage (wp.api.collections.Categories wrapper) with withAPIData HoC in Categories Block.

Categories block was the only block in our codebase that uses wp.api…, this change makes the block more inline with the rest of the codebase.

Commit:
Little polish to the top menu and inserter.

Commit:
Make mobile scroll "normally".

That is, scroll the `html` element, instead of a specially dedicated area. Although the latter has a ton of benefits, the former is the only thing that works stably on Mobile Safari. The thing is, there's overscroll bounce that you can't really turn off, so if you aren't scrolling the main content area things are just going to be all kinds of janky.

Commit:
Fix regression with "fixed to block" toolbar

Also rotate ellipsis so it matches top toolbar one.

Commit:
Prevent Safari from zooming in when opening inserter

youknowriad committed:
Move selectors and actions files to the store folder

aduth committed:
Factory: Fix block creation with falsey default

atimmer committed:
Add dynamic block documentation (#4067)

* Add dynamic block documentation

Taken from https://riad.blog/2017/10/16/one-thousand-and-one-way-to-extend-gutenberg-today/#dynamic-block.

* Add ES6 JavaScript code example

* Add a title to dynamic blocks doc entry

* Fix some small grammar errors

* Remove Riad from the code examples 😉

* Fix typo in dynamic block documentation

karmatosed committed:
3 column inserter

This is a suggestion for #4080. It increases the 2 columns to 3 for the inserter.

Commit:
Try improving mobile controls

Those controls include the mover and the ellipsis menu. As such, this PR fixes #4084.

What this PR does, is remove all mobile tools from the quick toolbar, and instead shows these tools inline at the bottom of a _selected_ block.

Commit:
Improve paragraph block description.

Props @maddisondesigns for text.

Commit:
Rename "Classic Text" to "Classic".

Fixes #2956.

This renames the "Classic Text" block to just "Classic", because it can do more than just text.

karmatosed committed:
Widens the inserter

Adds in a wider inserter.

karmatosed committed:
Left aligns the title

karmatosed committed:
Move title to center again

The left align wasn't working so lets move border and center align.

Commit:
Remove references to obsolete class.

youknowriad committed:
Block Mobile Toolbar: Extract as a separate component

youknowriad committed:
Merge branch 'try/improved-mobile-tools' of github.com:WordPress/gutenberg into try/improved-mobile-tools

ryo-utsunomiya committed:
Define setAttributes before using it.

jasmussen committed:
Merge pull request #4103 from WordPress/update/improve-paragraph-description

Improve paragraph block description.

jasmussen committed:
Merge pull request #4101 from WordPress/try/improved-mobile-tools

Try improving mobile controls

jorgefilipecosta committed:
Removed duplicate of buildTermsTree in HierarchicalTermSelector. (#4111)

Commit:
Merge pull request #4104 from WordPress/update/rename-classic-text

Rename "Classic Text" to "Classic". Props @jasmussen

Commit:
Merge pull request #4095 from WordPress/try/three-column-inserter

3 column inserter

Commit:
Polish permalink toolbar on mobile

- Add padding when fixed toolbar is enabled so the white sheet doesn't cover the permalink bar
- Fit the permalink bar snugly to the screen edges on mobile

jasmussen committed:
Merge pull request #4081 from WordPress/try/mobile-polish

Polish mobile, notably iPhone

youknowriad committed:
Chrome: Replace the publish dropdown by a sidebar panel

imath committed:
Use Post Type capabilities instead of All capabilities.

#3895 showed there was an issue with SuperAdmins not having a role on a site of a multisite network. After some discussions, we think it is best to check the post type capabilities for the connected user and the current post type being edited.
A new parameter is added to the users/me REST API route containing the post type name. This parameter is used to get the corresponding post type capabilities.
Capability checks have been updated accordingly into Gutenberg editor components

youknowriad committed:
Chrome: Allow multiple sidebars

youknowriad committed:
Chrome: Slide in animation for the publish sidebar

youknowriad committed:
Chrome: fix hasOpenSidebar selector

youknowriad committed:
Accessibility: Fix the accessibility of the menu items in the ellipsis menu

jorgefilipecosta committed:
Added drag&drop upload to cover image. (#4061)

Although cover-image block says "Drag image here or insert from media library", drag & drop upload was missing.

mcsf committed:
Heading <-> Cover Image transforms

youknowriad committed:
Chrome: Fix animation speed of the publish sidebar

jorgefilipecosta committed:
Made notices always visible; Added sticky behaviour in notices for widths lower than small breakpoint.

Used editor-left & editor-right mixins to correctly position notices.

youknowriad committed:
Code Block: Restore the placeholder

youknowriad committed:
Components: Allow explicitely overriding the ColorPalette colors per block

Commit:
Remove unnecessary line.

mcsf committed:
Merge pull request #4129 from WordPress/add/heading-cover-image-transforms

Add transforms between Heading & Cover Image

gziolo committed:
Remove unused variable from the_gutenberg_project function

gziolo committed:
Merge pull request #3937 from WordPress/try/edit-heading-round-2

Add editor heading for screenreaders

jorgefilipecosta committed:
Implemented editor-left&editor-right mixins.

The mixins apply editor left/right position to the selector passed as argument. It allows to use position fixed with correct dimensions.

youknowriad committed:
Gallery Block: Drag and drop multiple images create a gallery block

jorgefilipecosta committed:
Used editor-left mixin for .editor-header styles.

This change removes duplicate code and solves a bug related with resizing to small screen expanding WordPress menu and then resizing back to big screen.

youknowriad committed:
DropZone: Tweak its styling

BE-Webdesign committed:
Merge pull request #3894 from WordPress/add/test/editable/adapt-formatter

Add tests for Editable.adaptFormatter()

This is a private method, that will probably need to be reevaluated in how we test it.

BE-Webdesign committed:
Merge pull request #3892 from WordPress/add/test/blocks/editable-2

Add tests for Editable.getSettings()

This is a private method and the testing method may need to be reevaluated in the future.

getsource committed:
Fix typo in comment. (#4154)

s/dependecies/dependencies/

BE-Webdesign committed:
Merge pull request #3887 from WordPress/add/test/blocks/color-palette

Add tests for ColorPalette component.

youknowriad committed:
Block API: Adding a description property to the block API

youknowriad committed:
Blocks: Show the insertion point when the inserter is opened

vladanost committed:
Try: Add toggle editor mode shortcut (#3755)

youknowriad committed:
Chrome: Hide the Featured Image Panel if the CPT doesn't support it

youknowriad committed:
Chrome: Hide the Excerpt Panel if the CPT doesn't support it

youknowriad committed:
Chrome: Hide the discussion panel if no support for comments or trackbacks

youknowriad committed:
Chrome: Hide the PostAuthor selector if the CPT doesn't support it

youknowriad committed:
Chrome: User the PostTypeSupportCheck to check Post Formats support

youknowriad committed:
Chrome: Hide revisions if the CPT doesn't support it

ellatrix committed:
Handle copy for single block selection

youknowriad committed:
Mobile: Enhance the redux mobile enhancer to use the MediaQueries API

For more performance

youknowriad committed:
Blocks: Avoir reusable blocks styling

youknowriad committed:
Chrome: Hide The Taxonomies Panel if no taxonomy is available for the current CPT

youknowriad committed:
Taxonomiees: Improve Tags/Categories response size by limiting the requested fields

youknowriad committed:
UrlInput: Decode Post Title special chars

youknowriad committed:
Meta Boxes: When the metaboxes are saving ensure the overlay covers the whole area

laras126 committed:
Add text to meetup links

laras126 committed:
Organization and intitial population of articles page

noisysocks committed:
Add 'No saved blocks' message to Inserter

When there are no Reusable Blocks available to add, and the user selects
'Saved' in the Inserter, display a message instead of a completely blank
tab.

jorgefilipecosta committed:
Latest posts: Only request required category fields. (#4179)

We were requesting all category fields while needing three of them, this change makes sure we request only the fields we need.

travislopes committed:
Refactor `PostSchedule` to make calendar/clock available as components (#4180)

noisysocks committed:
Merge pull request #4182 from WordPress/add/no-saved-blocks-message

Add 'No saved blocks' message to Inserter

noisysocks committed:
Merge pull request #4137 from WordPress/add/reusable-block-deletion-to-api

Add ability to delete a Reusable Block to REST API

vladanost committed:
Check if mac when displaying editor mode toggle shortcut (#4194)

Rahmon committed:
Fix category in implementation of random image block

ntwb committed:
Update node-sass to v4.7.2  (#4200)

* Bump `node-sass` to latest 4.7.2 release

* Update `package-lock.json`

Soean committed:
Rebase
